gitlab/resource_gitlab_group_variable.go

```diff
@@ -1,12 +1,30 @@
 package gitlab
 
 import (
+	"fmt"
 	"log"
+	"strings"
 
+	"github.com/hashicorp/go-retryablehttp"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
 	gitlab "github.com/xanzy/go-gitlab"
 )
 
+// modifyRequestAddEnvironmentFilter returns a RequestOptionFunc function that
+// can be passed to the go-gitlab library calls to add the environment scope to
+// requests to lookup, modification, and deletion requests. Since gitlab 13.11,
+// an environment variable key is no longer unique and is composit-keyed with
+// the scope.
+// See https://docs.gitlab.com/ee/ci/variables/#add-a-cicd-variable-to-a-group
+func modifyRequestAddEnvironmentFilter(scope string) gitlab.RequestOptionFunc {
```

Comment:
Please use the context from the CRUD functions with `req.WithContext` here.

Reply:
I've updated the group_variables TF (258ceb8) definition [in accorance with the Terraform SDK v2 guide](https://www.terraform.io/plugin/sdkv2/guides/v2-upgrade-guide#more-support-for-context-context) to use the CRUD-with-context functions, and am now passing the context object into the gitlab HTTP requests (57f5be6).

I couldn't see this being done elsewhere in the project, so let me know if I've missed something and am doing it wrong.

Reply:
There is an open MR #791 introduced this to the entire code base.
Please rebase, and lgtm 🍰

Reply:
I have rebased against master. That MR looks like it might be merged relatively soon, if so should I wait and rebase again once it's in?

Reply:
I've rebased against master and tidied up now #791 is merged. I should have also fixed the linting error that was failing earlier

Reply:
@timofurrer Please, resolve this thread if you are ok with the current state. I will merge once you approve.

Reply:
I don't know why, but I can't resolve it :'(

Reply:
I have a resolve button... I'm happy we've resolved this conversation, but was wandering if the acceptance-ee test suite should be run before this gets merged?